src/thriveopedia/Thriveopedia.cs

```diff
@@ -0,0 +1,359 @@
+
+    /// <summary>
+    ///   Page history for backwards navigation.
+    /// </summary>
+    private Stack<ThriveopediaPage> pageHistory = new();
+
+    /// <summary>
+    ///   Page future for forwards navigation (if the player has already gone backwards).
+    /// </summary>
+    private Stack<ThriveopediaPage> pageFuture = new();
+
+    [Signal]
+    public delegate void OnThriveopediaClosed();
+
+    [Signal]
+    public delegate void OnSceneChanged();
+
+    /// <summary>
+    ///   The currently open Thriveopedia page. Defaults to the home page if none has been set.
+    /// </summary>
+    public ThriveopediaPage SelectedPage
+    {
+        get => selectedPage ?? homePage;
+        set
+        {
+            // Hide the last page and show the new page
+            selectedPage?.Hide();
+
+            selectedPage = value;
+            selectedPage.Show();
+
+            UpdateNavigationWithSelectedPage();
+        }
+    }
+
+    /// <summary>
+    ///   Details for the game currently in progress. Null if opened from the main menu. When set, initialises all
+    ///   pages requiring an active game.
+    /// </summary>
+    public GameProperties? CurrentGame
+    {
+        get => currentGame;
+        set
+        {
+            // We only want to initialise these pages once per game
+            if (value == currentGame)
+                return;
+
+            currentGame = value;
+
+            // Add all pages associated with a game in progress
+            if (currentGame != null)
+            {
+                AddPage("CurrentWorld");
+                AddPage("PatchMap", "CurrentWorld");
+                AddPage("EvolutionaryTree", "CurrentWorld");
+            }
+
+            // Notify all pages of the new game properties
+            foreach (var page in allPages.Keys)
+                page.CurrentGame = currentGame;
+        }
+    }
+
+    public override void _Ready()
+    {
+        backButton = GetNode<TextureButton>(BackButtonPath);
+        forwardButton = GetNode<TextureButton>(ForwardButtonPath);
+        pageContainer = GetNode<MarginContainer>(PageContainerPath);
+        pageTreeContainer = GetNode<PanelContainer>(PageTreeContainerPath);
+        pageTitle = GetNode<Label>(PageTitlePath);
+        pageTree = GetNode<Tree>(PageTreePath);
+
+        // Create and hide a blank root to avoid home being used as the root
+        pageTree.CreateItem();
+        pageTree.HideRoot = true;
+
+        // Keep a special reference to the home page
+        homePage = GetNode<ThriveopediaHomePage>(HomePagePath);
+        allPages.Add(homePage, CreateTreeItem(homePage, null));
+
+        // Add all pages not associated with a game in progress
+        AddPage("Museum");
+
+        pageHistory.Push(homePage);
+        SelectedPage = homePage;
+    }
+
+    public override void _Notification(int what)
+    {
+        base._Notification(what);
+
+        if (what == NotificationVisibilityChanged && Visible)
```

Comment:
Well we don't have translations yet but I think some places like the navigation tree needs to be regenerated on language change, I'll open a follow up issue for this as well.